File: SOA_Project/SOA_Project.scala
import org.apache.spark.sql.types.{IntegerType,StringType,StructType,StructField}
import org.apache.spark.sql._
import java.io._

object SOA_Project {
  def dataSetProcessing (row: Row, result: DataFrame, spark: SparkSession, resultSchema: StructType){
    val newRow = List[Row](row.getAs("image"))



    val rddObj = spark.sparkContext.parallelize(newRow)
    result.union(rddObj)
  }

  def main(args: Array[String]) {
    //Definizione delle sessione spark
    val spark = SparkSession.builder.appName("Test_Application").getOrCreate()

    //Caricamento DataSet JSON
    val path = "Arienzo-Giordano-Scotti/DataSetPic.json"
    val dataset = spark.read.json(path)
    
    //Definizione schema da DataFrame finale
    val resultSchema = StructType(Array(
      StructField("image",StringType,true),
    ))

    //Creazione dal DataFrame vuoto
    val result = spark.createDataFrame(spark.sparkContext.emptyRDD[Row],resultSchema)

    println()
    println("Inizio output programma: ")
    println()

    //Iterazione del DataFrame caricato
    dataset.foreach(row => dataSetProcessing(row, result, spark, resultSchema))

    //Scrittura del DataFrame finale su HDFS
    result.write.json("Arienzo-Giordano-Scotti/result.json")

    println()
    println("Fine output programma:")
    println()

    spark.stop()
  }
}

File: src/main/java/SOA_Project.java
import java.util.ArrayList;
import java.util.List;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Encoder;
import org.apache.spark.sql.Encoders;

import org.apache.spark.sql.SparkSession;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;

public class SOA_Project {
  
  public static void dataSetProcessing(DataSetRow row, List<ResultRow> images){
    images.add(new ResultRow(row.getImage()));
  } 

  public static void main(String[] args) {
    SparkConf conf = new SparkConf().setAppName("SOA Project").setMaster("yarn");
    JavaSparkContext sc = new JavaSparkContext(conf);
    SparkSession spark = SparkSession.builder().getOrCreate();

    //Creazione java bean per la definizione dello schema del DataSet
		Encoder<DataSetRow> dataSetEncoder = Encoders.bean(DataSetRow.class);
    Encoder<ResultRow> resultEncoder = Encoders.bean(ResultRow.class);
    String path = "Arienzo-Giordano-Scotti/DataSet.json";

    Dataset<DataSetRow> dataset = spark.read().json(path).as(dataSetEncoder);

    List<ResultRow> images = new ArrayList<ResultRow>();

    dataset.toJavaRDD().collect().forEach(row -> images.add(new ResultRow(row.getImage())));

    //images.add(new ResultRow("ciaoooooo c'è nessuno ?"));
    //images.add(new ResultRow(dataset.first().getImage()));

    Dataset<ResultRow> result = spark.createDataset(images, resultEncoder);

    result.repartition(1).write().format("json").save("Arienzo-Giordano-Scotti/result.json");

    sc.close();
    spark.stop();
  }
}
